### PR TITLE
ADDED: Undead Rogue Shadowstrike Rune Quest

### LINKED ISSUE #5298: Missing Quest for Undead Rogues
## Description
Quest in tracker not present in QuestDB!

Quest ID: 77669
Quest Name: The Scarlet Rune
Quest Text: Who were you in life, <playername>? A spy? A thief? A cook who was especially handy with a knife? Whatever your past, it's time to embrace new tactics. The rogues of the Forsaken seize every advantage offered to them.

The Scarlet Converts to the east fight like they've never seen combat, but the Crusade has issued them mystical runes to enhance their abilities. Steal one of these runes and learn its magic, then return to me.
Objective Text: Pickpocket or kill Scarlet Converts to find a rune, then use it to learn a new ability. Afterwards, return to David Trias in Deathknell.
Character: Lvl 12 SCOURGE ROGUE
Player Coords:  [1420]  (32.561, 64.922)
QuestLog: 5723, 8368, 408, 356, 360, 5724, 357, 371, 369, 7024, 77669, 450, 
Client: 1.15.0 SoD
Questie: v9.2.2
Locale: enUS

## Questie & WoW version
Season of Discovery
Questie 9.2.2

### PR DESCRIPTION
**Pull request by Sommos**
## Issue references

Fixes #5298

## Proposed changes

- ADDED: `The Scarlet Rune` quest to the database, should now be accessible to users.

### CODE REVIEW

File: Database/Corrections/Automatic/sodBaseQuests.lua [hunk 1]
```diff
@@ -181,6 +181,17 @@ function SeasonOfDiscovery:LoadBaseQuests()
...
+        [77669] = {
+            [questKeys.name] = "The Scarlet Rune",
+            [questKeys.startedBy] = {{2122}},
+            [questKeys.finishedBy] = {{2122,}},
+            [questKeys.requiredLevel] = 2,
+            [questKeys.questLevel] = 2,
+            [questKeys.requiredRaces] = raceIDs.ALL_HORDE,
+            [questKeys.requiredClasses] = classIDs.ROGUE,
+            [questKeys.objectivesText] = {"Pickpocket or kill Scarlet Converts to find a rune, then use it to learn a new ability. Afterwards, return to David Trias in Deathknell."},
+            [questKeys.objectives] = {nil,nil,{{400105}}},
```
Comment (BreakBB) [inline Database/Corrections/Automatic/sodBaseQuests.lua:193]:
```suggestion
            [questKeys.objectives] = {nil,nil,nil,nil,nil,{{400105}}},
```

Even though this field is overridden in the `sodBaseQuestFixes`, the objectives should be valid in a valid structure.

Explanation: You added the "spellObjective" (spell 400105) as "itemObjective". But there is no item "400105".

File: Database/Corrections/sodQuestFixes.lua [hunk 2]
```diff
@@ -100,6 +100,12 @@ function SeasonOfDiscovery:LoadQuests()
             [questKeys.zoneOrSort] = sortKeys.WARLOCK,
             [questKeys.requiredSpell] = -403919,
         },
+        [77669] = {
+            [questKeys.objectives] = {nil, nil, nil, nil, nil, {{410002, nil, 205420}}},
+            [questKeys.requiredRaces] = raceIDs.UNDEAD,
+            [questKeys.zoneOrSort] = sortKeys.ROGUE,
+            [questKeys.requiredSpell] = -410002,
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:107]:
```suggestion
            [questKeys.requiredSpell] = -400105,
```

This needs to be the [Engrave](https://www.wowhead.com/classic/spell=400105/engrave-gloves-shadowstrike) version of the spell.

**Review by BreakBB — CHANGES_REQUESTED** [review]
Only some really minor changes required. Great job again @Sommos 🥳

**Comment by Sommos** [comment]
I'll get the hang of it as I PR more and more, and refine these mistakes out. Just bear with me. 🥳

**Comment by BreakBB** [comment]
No worries at all @Sommos - Again, we appreciate your help and feel free to ask if something is unclear or seems _fishy_. You can also hit us up on Discord, most team members are better available over there.

**Review by BreakBB — APPROVED** [review]

**Pull request merged**